lib/draft/feature_test_phoenix_integration.md

```diff
@@ -0,0 +1,232 @@
+    conn: conn,
+    student: student,
+    contract: contract,
+    user: user
+  } do
+    conn
+    |> get(Routes.pow_session_path(conn, :new))
+    |> follow_form(%{
+      user: %{
+        email: user.email,
+        password: user.password
+      }
+    })
+    |> get(Routes.student_path(conn, :index))
+    |> follow_link(student.first_name)
+    |> assert_response(
+      status: 200,
+      path: Routes.student_path(conn, :show, student),
+      html: "Contracts",
+      body: contract.start_date |> to_string
+    )
+  end
+end
+```
+
+So let's see what is going on here. You'll notice that my setup is pretty extensive. That is basically because this test is about seeing the list of contracts on the student detail page. Which means we'll need to have some content to be able to see it. In addition to this test, we could also make some variations that test that we'll show a message to indicate that there are no contracts when there are none added yet.
+
+## Testing with authenticated users
+
+I wanted to add a sidenote regarding testing with authenticated users. When I tested features with [Wallaby](https://github.com/elixir-wallaby/wallaby) I always used the method provided by [Pow](https://github.com/danschultzer/pow#testing-with-authenticated-users) to set the current user in the assigns like this:
+
+```elixir
+setup %{conn: conn} do
+  user = %User{email: "test@example.com"}
+  conn = Pow.Plug.assign_current_user(conn, user, otp_app: :my_app)
+
+  {:ok, conn: conn}
+end
+```
+
+Now I started out that same way and couldn't get it to work. Apparently when I made two consequent requests in my test pipeline, the assigns would get wiped. At first I was unsure if it was a bug or if it was just something I was doing wrong. So I added an issue to discuss it with Boyd(the creator of Phoenix Integration), and he did some digging and explained fully what was happening. You can see that discussion [here](https://github.com/boydm/phoenix_integration/issues/38). So the reason that just setting the current assign won't work, is because `Phoenix.ConnTest` actually wipes the assigns between each request. Basically mimicing stateless browser request behaviour. They call this recycling and is explained [here](https://github.com/phoenixframework/phoenix/blob/9f703bfec83a2f35102af7c9489ebb4a01c35f6a/lib/phoenix/test/conn_test.ex#L79). They do however carry over cookies that are set, so basically the solution was to actually login through the browser in my test.
+
+In the example above I am login in directly in the pipeline, but you really don't want to do this all over the place if you add a lot of tests. In the [github issue](https://github.com/boydm/phoenix_integration/issues/38#issuecomment-593679075) Boyd also gives an example of how he does it mostly. Basically you can add a helper function in `ConnCase` you can call to sign in different types of users easily. For my example I'll leave it as because this post is long enough already.
```

Comment:
> how he does it mostly

> how he does this mostly

Reply:
> For my example I'll leave it as

> For my example I'll leave this as is and

----
>  because this post is long enough already.

That is perception, but I. can tell you that your blog post reads nicely, and I do not have the feeling that this is too long (on a sidenote)